napari/_qt/_qapp_model/_tests/test_view_menu.py

```diff
@@ -50,28 +51,114 @@ def test_toggle_axes_scale_bar_attr(
 
 
 @skip_local_popups
+@pytest.mark.qt_log_level_fail('WARNING')
 def test_toggle_fullscreen(make_napari_viewer, qtbot):
-    """Test toggle fullscreen action."""
+    """
+    Test toggle fullscreen action.
+
+    Check that toggling from a normal and maximized state can be done without
+    generating any type of warning and menu bar elements are visible in any
+    window state.
+    """
     action_id = 'napari.window.view.toggle_fullscreen'
     app = get_app_model()
     viewer = make_napari_viewer(show=True)
 
+    def check_windows_style():
+        if os.name != 'nt':
+            return
+        import win32con
+        import win32gui
+
+        window_handle = viewer.window._qt_window.windowHandle()
+        window_handle_id = int(window_handle.winId())
+        window_style = win32gui.GetWindowLong(
+            window_handle_id, win32con.GWL_STYLE
+        )
+        assert window_style & win32con.WS_BORDER == win32con.WS_BORDER
```

Comment:
Is there reason to make this inline function?
I prefer to make it top-level function and pass `viewer` as argument.

Reply:
Thanks for the new feedback! I moved the function to be a top-level one and also divided the test into two tests (one testing things from a normal window state and one testing things from a maximized window state). Let me know if there is something else to improve here!